File: Data Structures & Algorithms/rotate-array/submission-3.java
class Solution {
    public void rotate(int[] nums, int k) {
        //   0 1 2 3 4 5 6 
        //   1 2 3 4 5 6 7
        //   i     k 
    //temp:  4 -3 
        int[] temp = new int[nums.length];
        for(int i = 0; i < temp.length; i++){
            if(k < nums.length){
                temp[k] = nums[i];
                k++;
            }else if(k == nums.length){
                k = 0;
                temp[k] = nums[i];
                k++;
            }else if(k > nums.length){
                k = k % nums.length;
                temp[k] = nums[i];
                k++;
            }

        }
        for(int i = 0; i < temp.length; i++){
            nums[i] = temp[i];
        }
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        #       2
        #      2 5
        res = []
        currSum = 0
        stack = []

        def backtrack(pos):
            nonlocal res, currSum, stack
            if currSum == target:
                res.append(stack[:])
                return
            elif currSum > target or pos == len(nums):
                return
            currSum += nums[pos]
            stack.append(nums[pos])
            backtrack(pos)
            currSum -= stack.pop()
            backtrack(pos+1)

        backtrack(0)
        return res

            
                